lighthouse-core/gather/gatherers/dobetterweb/console-time-usage.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Tests whether the page is using console.time().
+ */
+
+/* global window, __returnResults */
+
+'use strict';
+
+const Driver = require('../../drivers/driver.js');
+const Gatherer = require('../gatherer');
+
+function collectUsage() {
+  __returnResults(Array.from(
+      window.__consoleTimeStackTraces).map(item => JSON.parse(item)));
+}
+
+class ConsoleTimeUsage extends Gatherer {
+
+  beforePass(options) {
+    const driver = options.driver;
+    return driver.evaluateScriptOnLoad(`
+        window.__consoleTimeStackTraces = new Set();
```

Comment:
this is still a bit fiddly on the caller's side.

Could more of this be pulled into driver? Maybe `captureAPIUsage` could set up the collecting as it does here but also handles the `evaluateScriptOnLoad` part and returns a `collectUsage` function that could be called whenever (probably usually in `afterPass`) that takes care of fetching the array of callsites for you? all without having to know that you're evaluating a script, that the callsites are on `__consoleTimeStackTraces` in the host page, etc